src/caffeine/prif_private_s.f90

```diff
@@ -0,0 +1,10 @@
+! Copyright (c), The Regents of the University of California
+! Terms of use are as specified in LICENSE.txt
+submodule(prif) prif_private_s
+  implicit none
```

Comment:
The Fortran spec rules seem potentially ambiguous regarding the default accessibility behavior of a submodule.

Let's add an explicit private here, both to ensure all compilers treat this submodule's data as private to the `prif` implementation and for documentation purposes.

```suggestion
  implicit none
  private
```

Reply:
@bonachea Section `8.5.2 Accessibility attribute` of the Fortran 2023 Standard has a constraint that disallows this.
`C817 An access-spec shall appear only in the specification-part of a module.`

Reply:
> The Fortran spec rules seem potentially ambiguous regarding the default accessibility behavior of a submodule.

Only entities declared in a module are accessible via a `use` of that module. All submodule contents are private implementation details.

Reply:
> Only entities declared in a module are accessible via a `use` of that module. All submodule contents are private implementation details.

Thanks for the confirmation Brad. I suspected this was the case, but could not find confirmation on a brief skim of the relevant spec sections.